src/eval.ml

```diff
@@ -210,7 +202,13 @@ and tree_walk : Dtree.t -> int -> term list -> (term * term list) option =
                       then raise Unallowed) to_stamped; true
                 with Unallowed -> false
               in
-              if !log_enabled then log_fvc r xs;
+              if !log_enabled then
+                begin
+                  let module F = Format in
```

Comment:
Note that we have a `Array.pp` function that should work here. You can replace the whole block with the following, and readability will be improved again.
```ocaml
log_eval (r_or_g r "Free var check on [%a]") (Array.pp pp_tvar "; ") xs
```